### PR TITLE
fix: Blob.stream return type which is incorrect and conflicts with lib.dom.d.ts

### PR DESCRIPTION
**Pull request by Gozala**
I frequently run into following issue 

image

Caused by incorrect type annotation of Blob

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2280eb45b474bf8e92ff34b1d2775afc03cffc6f/types/node/stream/consumers.d.ts#L11

Proposed changes replace that annotation with a more accurate one so that it:

1. Does not conflict with `lib.dom.d.ts`
2. Reflects actual state of things in node & specifically the fact that `Blob.prototype.stream` returns web stream and not a node one. See https://nodejs.org/api/buffer.html#blobstream

I do not know how current typedef ended this way. I think comment attempt to elaborate reason for this `Blob` interface definition, yet it does not explain why `stream()` method is typed incorrectly.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by Gozala** [comment]
Looks like issue was raised during review https://github.com/DefinitelyTyped/DefinitelyTyped/pull/55311#discussion_r787646125 but things have landed anyhow.

**Comment by Gozala** [comment]
@favna I think you've introduced this definition into codebase, mind providing some context here and whether proposed changes would cause problems ?

Thanks

**Comment by Gozala** [comment]
Looks like `ReadableStream` is defined in `lib.dom.d.ts` which as per comment should not be referenced otherwise failing tests would pass as well.

Again I'm not sure I understand the intricacies here, only other alternative I can think of is removing `stream` method all together so it does not conflict with `lib.dom.d.ts` yet it feels like proper fix here would be to reference actual `Blob` instead.

**Comment by favna** [comment]
@Gozala The comment you linked was raised after the PR was already approved (by DT maintainers mind you, which is required for node types) and merged, ergo "it was merged anyhow"... Doesn't really apply. 

Obviously we cannot fully remove stream because it is actually functionality that's in NodeJS: https://nodejs.org/dist/latest-v17.x/docs/api/stream.html

I'm sure the maintainers of Node are all open to potential solutions, but I'm not one of them by choice so all of this said I will distance myself from this issue.

**Comment by favna** [comment]
Addition to my comment above, I recommend you undraft this PR. Right now no one *really* becomes aware that the PR exists. Only I did because you mentioned me. However once you open the PR for review the GitHub bot will ping all maintainers and then people can let you know what's what.

**Comment by favna** [comment]
> > Obviously we cannot fully remove stream because it is actually functionality that's in NodeJS: [nodejs.org/dist/latest-v17.x/docs/api/stream.html](https://nodejs.org/dist/latest-v17.x/docs/api/stream.html)
> 
> I'm not sure I fully understand the relation here. I'd like to remove type incompatibility between definitions, not remove the stream definition.

I think I just misread one of your messages. Ignore that of mine.

**Comment by LinusU** [comment]
Could a potential fix be something like this:

```diff
+ import { ReadableStream } from 'node:stream/web'
  // Duplicates of interface in lib.dom.ts.
  // Duplicated here rather than referencing lib.dom.ts because doing so causes lib.dom.ts to be loaded for "test-all"
  // Which in turn causes tests to pass that shouldn't pass.
  //
  // This interface is not, and should not be, exported.
  interface Blob {
      readonly size: number;
      readonly type: string;
      arrayBuffer(): Promise<ArrayBuffer>;
      slice(start?: number, end?: number, contentType?: string): Blob;
-     stream(): NodeJS.ReadableStream;
+     stream(): ReadableStream;
      text(): Promise<string>;
  }
```

This would correct the typings, and then hopefully be compatible with the typings in `dom.d.ts`?

**Review by gabritto — CHANGES_REQUESTED** [review]
so, from what I understood about the comments: we don't really want to remove `stream()`, but instead change the return type somehow, right? @LinusU left a comment suggesting a possible approach: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59057#issuecomment-1056815186
you could try that, and possibly update tests (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f4c3955fe966c65fe528c1963e0fa217034c6ca5/types/node/test/stream.ts), and see if it fixes your issue and doesn't break anything.

File: types/node/stream/consumers.d.ts
Comment (Gozala) [inline types/node/stream/consumers.d.ts:11]:
```suggestion
    stream(): import('node:stream/web').ReadableStream;
    text(): Promise<string>;
```

**Comment by Gozala** [comment]
This somehow never end up in my notifications. I have created new pull request https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60377 which does what @LinusU has suggested here
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59057#issuecomment-1056815186

I'm not familiar with how things are wired here so I have no idea if this would create some other issues. Either way I think it would be best to remove `stream()` method as:

1. It is simply incorrect
2. Causes problems caused by conflicts with built in typedefs.
3. No tests seem to break removing it presumably because nothing actually depends on it.

If there is a desire to have more correct `stream()` method that could be pursued separately as an improvement. First change would still resolve incompatibility issues without blocking on figuring out all the quirks of how outlined here

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/063f0c8cedb9458e4e07606fc3ba1793ff0ab705/types/node/stream/consumers.d.ts#L1-L5

Not to mention the fact that only reason this type definition seems to be around is to avoid loading `Blob` definition form `dom.d.ts` which I'm still not sure I understand why is that desired.

**Pull request closed without merging**
